app/src/test/adminUsers.test.tsx

```diff
@@ -1,12 +1,421 @@
+    emailAddress: 'user2@ualberta.ca',
+    inUse: 2,
+    overdue: 5,
+    role: 'admin',
+  },
+  {
+    userId: 'user3',
+    emailAddress: 'mysterious@ualberta.ca',
+    inUse: 3,
+    overdue: 39,
+    role: 'volunteer',
+  },
+  {
+    userId: 'user4',
+    emailAddress: 'mocked-email@ualberta.ca',
+    inUse: 5,
+    overdue: 45,
+    role: 'admin',
+  }
+];
+
+const mockInUseResponse = [
+  {
+    email: 'hello@ualberta.ca',
+    count: 2
+
+  },
+  {
+    email: 'user2@ualberta.ca',
+    count: 2
+
+  },
+  {
+    email: 'mysterious@ualberta.ca',
+    count: 3
+
+  },
+  {
+    email: 'mocked-email@ualberta.ca',
+    count: 5
+  }
+  
+];
+
+const mockOverdueResponse = [
+  {
+    email: 'hello@ualberta.ca',
+    count: 30
+
+  },
+  {
+    email: 'user2@ualberta.ca',
+    count: 5
+
+  },
+  {
+    email: 'mysterious@ualberta.ca',
+    count: 39
+
+  },
+  {
+    email: 'mocked-email@ualberta.ca',
+    count: 45
+  }
+]
+
+const mockDishes = [
+  {
+    id: 1,
+    qid: 2,
+    registered: '2023-07-22T20:19:47.144Z',
+    status: 'available',
+    timesBorrowed: 7,
+    type: "plate",
+    userId: "yo"
+
+  }
+]
+
+const mockUsers = [
+  {
+    userId: 'user1',
+    emailAddress: 'hello@ualberta.ca',
+    role: 'customer',
+  },
+  {
+    userId: 'user2',
+    emailAddress: 'user2@ualberta.ca',
+    role: 'admin',
+  },
+  {
+    userId: 'user3',
+    emailAddress: 'mysterious@ualberta.ca',
+    role: 'volunteer',
+  },
+  {
+    userId: 'user4',
+    emailAddress: 'mocked-email@ualberta.ca',
+    role: 'admin'
+  }
+]
+
+const mockModifyRole = {
+  data:{
+  status: "Success"
+  }
+}
+
+
+beforeEach(async () => {
+    jest.clearAllMocks(); // Clears any previous mocking behavior before each test
+
+    //mocking all adminApi functions which are called in the flow of the admin users page
+    jest.mock('../admin/adminApi', () => ({
+      getDishesStatusForEachUser: jest.fn(() => Promise.resolve(mockDishesStatusData)),
+      getAllDishes: jest.fn(() => Promise.resolve(mockDishes)),
+      getUsers: jest.fn(() => Promise.resolve(mockUsers)),
+      modifyRole: jest.fn(() => Promise.resolve({data:{status: 'Success'}})),
+
+    }));
+
+    //Mock response to be returned by our mock implementation of the useAuth
+    useAuthMock.mockImplementation(() => ({
+      currentUser: {
+        id: 'mocked-user-id',
+        role: 'admin',
+        email: 'mocked-email@ualberta.ca'
+      },
+      sessionToken: 'mocked-session-token',
+      login: jest.fn(),
+      logout: jest.fn(),
+    }));
+
+  
+});
+
+describe("Renders properly", () => {
+  it("it renders Admin Users Page without crashing", () => {
+    render(
+      <BrowserRouter>
+        <Users />
+      </BrowserRouter>);
+  })})
+
+describe('Table/Mainframe Check', () => {
+  it('renders the four column headers', async () => {
+    
+    await act(async () => {
+      render(
+        <BrowserRouter>
+          <Users />
+        </BrowserRouter>
+      );
+    });
```

Comment:
I might be wrong, but I don't think you need to wrap this in an `await act()`. `act` is usually when you need to perform a user action (like a click or something). does it still work if you remove it?

```suggestion
      render(
        <BrowserRouter>
          <Users />
        </BrowserRouter>
      );
```

Reply:
it doesn't. It forced me to put an act. then the error went away